framework/src/userobjects/ElementSubdomainModifier.C

```diff
@@ -544,98 +739,228 @@ ElementSubdomainModifier::pushBoundaryNodeInfo(MooseMesh & moose_mesh)
+        {
+          nearest_distance[activated_elem->id()] = dist;
+          dofmap.dof_indices(elem, _elem_nearest_dofs[var_num][activated_elem->id()], var_num);
+        }
+      }
+      else
+      {
+        nearest_distance[activated_elem->id()] = std::numeric_limits<Real>::max();
+        dofmap.dof_indices(elem, _elem_nearest_dofs[var_num][activated_elem->id()], var_num);
+      }
+    }
+  }
+}
+
+void
+ElementSubdomainModifier::setNearestSolutionForActivatedDofs(SystemBase & sys,
+                                                             const VariableName & var_name,
+                                                             NumericVector<Real> & old_solution)
+{
+  // Get the variable
+  auto & var = sys.getVariable(_tid, var_name);
+
+  // Get the variable number
+  auto var_num = var.number();
+
+  auto & current_solution = sys.solution();
   DofMap & dof_map = sys.dofMap();
 
-  // Get dofs for the newly added elements
   std::vector<dof_id_type> dofs;
-  for (auto & bnd_node : bnd_node_range)
+  std::vector<dof_id_type> nearest_dofs;
+
+  if (_node_nearest_dofs.count(var_num))
   {
-    std::vector<dof_id_type> bnd_node_dofs;
-    dof_map.dof_indices(bnd_node->_node, bnd_node_dofs);
-    dofs.insert(dofs.end(), bnd_node_dofs.begin(), bnd_node_dofs.end());
+    for (auto & [activated_node_id, nearest_node_dofs] : _node_nearest_dofs[var_num])
+    {
+      auto activated_node = _mesh.nodePtr(activated_node_id);
+      // Dofs on the activated node
+      std::vector<dof_id_type> node_dofs;
+      dof_map.dof_indices(activated_node, node_dofs, var_num);
+      dofs.insert(dofs.end(), node_dofs.begin(), node_dofs.end());
+
+      // Dofs on the nearest node
+      nearest_dofs.insert(nearest_dofs.end(), nearest_node_dofs.begin(), nearest_node_dofs.end());
+    }
   }
 
-  // Set the old and older solution to match the IC.
-  for (auto dof : dofs)
+  if (_elem_nearest_dofs.count(var_num))
   {
-    old_solution.set(dof, current_solution(dof));
-    if (older_solution)
-      older_solution->set(dof, current_solution(dof));
+    for (auto & [activated_elem_id, nearest_elem_dofs] : _elem_nearest_dofs[var_num])
+    {
+      auto activated_elem = _mesh.elemPtr(activated_elem_id);
+      // Dofs on the activated element
+      std::vector<dof_id_type> elem_dofs;
+      dof_map.dof_indices(activated_elem, elem_dofs, var_num);
+      dofs.insert(dofs.end(), elem_dofs.begin(), elem_dofs.end());
+
+      // Dofs on the nearest element
+      nearest_dofs.insert(nearest_dofs.end(), nearest_elem_dofs.begin(), nearest_elem_dofs.end());
+    }
   }
 
+  for (auto i : index_range(dofs))
+    current_solution.set(dofs[i], old_solution(nearest_dofs[i]));
+
+  current_solution.close();
   old_solution.close();
-  if (older_solution)
-    older_solution->close();
+}
+
+void
+ElementSubdomainModifier::setConstantForActivatedDofs(SystemBase & sys,
+                                                      const VariableName & var_name)
+{
```

Comment:
Don't see anything in this function that would keep it from being used for
`Steady` problems.